app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb

```diff
@@ -8,33 +8,42 @@ def initialize(_manager, _target)
     target.manager_refs_by_association_reset
   end
 
-  def clusters
-    if target.kind_of? VmOrTemplate
-      return collect_emsclusters
+  def ems_clusters
+    clusters = []
+    if target.targets.any? { |t| t.kind_of? VmOrTemplate }
```

Comment:
right, so for the saving to work properly, the Collector and Persistor should use the same: references(:ems_clusters)

references(:ems_clusters) => [:cluster_ems_ref_1, :cluster_ems_ref_2]
So what it will then say, is collect me only  [:cluster_ems_ref_1, :cluster_ems_ref_2] from the API. And when saving, compare it to [:cluster_ems_ref_1, :cluster_ems_ref_2] in the MIQ DB.

so Collector and Persistor should be aligned like this